fmn/consumer/backend.py

```diff
@@ -152,22 +152,33 @@ def read(queue_object):
 
     if function:
         print("Got a function call to %s" % function)
-        if function == 'handle':
-            backend = backends[data['backend']]
-            backend.handle(session,
-                           data['recipient'],
-                           data['msg'],
-                           data['streamline'])
-            ch.basic_ack(delivery_tag=method.delivery_tag)
-        elif function == 'handle_batch':
-            backend = backends[data['backend']]
-            backend.handle_batch(session,
-                                 data['recipient'],
-                                 data['queued_messages'])
+        try:
+            if function == 'handle':
+                backend = backends[data['backend']]
+                backend.handle(session,
+                               data['recipient'],
+                               data['msg'],
+                               data['streamline'])
+                ch.basic_ack(delivery_tag=method.delivery_tag)
+            elif function == 'handle_batch':
+                backend = backends[data['backend']]
+                backend.handle_batch(session,
+                                     data['recipient'],
+                                     data['queued_messages'])
+                ch.basic_ack(delivery_tag=method.delivery_tag)
+            else:
+                print("Unknown function")
+            return
+        except Exception:
+            logging.error('Message failed, requeueing', exc_info=True)
+            ch.basic_publish(exchange='',
```

Comment:
As discussed on IRC: I am now manually requeueing to put it back at the end of the queue.
The basic_nack puts it back at the front of the queue, still leaving the worker unable to send out any messages until this message is fixed.

I'm going to say no to dropping messages entirely in general.

We should note that this PR makes it so that messages might be sent out in a different order then received, but to a certain extent that'll always be the case with multiple workers.